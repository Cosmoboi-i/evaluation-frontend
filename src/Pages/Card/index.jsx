import React from "react";
import { useParams } from "react-router-dom";
import { Main } from "../../components";

export default function Card() {
  const { id } = useParams();
  return <Main id={id} />;
}
