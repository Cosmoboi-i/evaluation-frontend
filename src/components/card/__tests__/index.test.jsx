import React from "react";
import { render } from "@testing-library/react";
import Card from "..";

export const mockCard = {
  id: 1,
  name: "abc",
  description: "abc",
  venue: "abc",
  datetime: "abc",
  timezone: "abc",
  areSeatsAvailable: false,
  isRegistered: false,
  isBookmarked: false,
  imgUrl: "ad",
  updateEvent: jest.fn(),
};

describe("Card", () => {
  it("should render properly", () => {
    const { asFragment } = render(<Card {...mockCard} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
