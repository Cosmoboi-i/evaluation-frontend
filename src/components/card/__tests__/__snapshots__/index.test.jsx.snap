// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render properly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <img
      src="ad"
    />
    <div
      class="card-content"
    >
      <div
        class="card-heading"
      >
        abc
      </div>
      <div
        class="card-content"
      >
        <div
          class="card-desc"
        >
          abc
        </div>
        <div>
          <div
            class="venue"
          >
            <b>
              VENUE:
            </b>
             abc
          </div>
          <div
            class="venue"
          >
            <b>
              DATE:
            </b>
             abc
          </div>
        </div>
      </div>
      <div
        class="reactions"
      >
        <div
          class="register no-seats"
        >
          <i
            class="fa-solid fa-circle-xmark fa-2x"
          />
          <div
            class="registered"
          >
            NO SEATS AVAILABLE
          </div>
        </div>
        <i
          class="fa-regular fa-bookmark fa-2x bookmark"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
